doc/performance.md

```diff
@@ -0,0 +1,97 @@
+# Performance with Native Code
+
+How to assess performance of Dart and native code, and how to improve it.
+
+## Profiling Performance
+
+
+| Tool                                  | Platform  | Primary Use Case                        | Measures (Dart CPU)          | Measures (Native CPU)    | Measures (Dart Heap) | Measures (Native Heap)                                           |
```

Comment:
Optional: It may be helpful to have the tool names link to the tool's websites where people can learn more about these tools in general.